kernel/kernel-defaults/src/test/scala/io/delta/kernel/defaults/DeltaClusteringSuite.scala

```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright (2025) The Delta Lake Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.delta.kernel.defaults
+
+import scala.collection.JavaConverters._
+import scala.collection.immutable.Seq
+
+import io.delta.kernel.Operation.{CREATE_TABLE, WRITE}
+import io.delta.kernel.Table
+import io.delta.kernel.exceptions.TableAlreadyExistsException
+import io.delta.kernel.expressions.{Column, Literal}
+import io.delta.kernel.expressions.Literal.ofInt
+import io.delta.kernel.internal.SnapshotImpl
+import io.delta.kernel.internal.actions.DomainMetadata
+import io.delta.kernel.internal.clustering.ClusteringMetadataDomain
+import io.delta.kernel.types.IntegerType.INTEGER
+import io.delta.kernel.types.StructType
+import io.delta.kernel.utils.CloseableIterable.emptyIterable
+
+class DeltaClusteringSuite extends DeltaTableWriteSuiteBase {
+
+  test("create clustered table - clustering column is not part of the schema") {
+    withTempDirAndEngine { (tablePath, engine) =>
+      val table = Table.forPath(engine, tablePath)
+      val txnBuilder = table.createTransactionBuilder(engine, testEngineInfo, CREATE_TABLE)
+
+      val ex = intercept[IllegalArgumentException] {
+        val txn = txnBuilder
+          .withSchema(engine, testPartitionSchema)
+          .withClusteringColumns(engine, List(new Column("PART1"), new Column("part3")).asJava)
+          .build(engine)
+        commitTransaction(txn, engine, emptyIterable())
```

Comment:
can you update the `createEmptyTable` to take clustering as one of the optional args and use that here?